superset-frontend/cypress-base/cypress/integration/dashboard/edit_mode.test.js

```diff
@@ -23,47 +23,54 @@ describe('Dashboard edit mode', () => {
     cy.server();
     cy.login();
     cy.visit(WORLD_HEALTH_DASHBOARD);
-    cy.get('.dashboard-header [data-test=edit-alt]').click();
+    cy.get('[data-test="dashboard-header"]')
+      .find('[data-test=edit-alt]')
+      .click();
   });
 
-  xit('remove, and add chart flow', () => {
+  it('remove, and add chart flow', () => {
     // wait for box plot to appear
-    cy.get('.grid-container .box_plot');
-
-    cy.get('.fa.fa-trash')
+    cy.get('[data-test="grid-container"]').find('.box_plot', {
+      timeout: 10000,
+    });
+    cy.get('[data-test="dashboard-component-chart-holder"]')
+      .find('[data-test="dashboard-delete-component-button"]')
       .last()
       .then($el => {
         cy.wrap($el).invoke('show').click();
         // box plot should be gone
-        cy.get('.grid-container .box_plot').should('not.exist');
+        cy.get('[data-test="grid-container"]')
+          .find('.box_plot')
+          .should('not.be.visible');
       });
 
-    cy.get('.tabs-components .nav-tabs li a').contains('Charts').click();
+    cy.get('[data-test="dashboard-builder-component-pane-tabs-navigation"]')
+      .children()
+      .last()
+      .click();
 
     // wait for tab-switching animation to complete
-    cy.wait(1000);
+    cy.wait(2000);
 
     // find box plot is available from list
-    cy.get('.tabs-components')
-      .find('.chart-card-container')
-      .contains('Box plot');
+    cy.get('[data-test="card-title"]').contains('Box plot', { timeout: 5000 });
 
-    drag('.chart-card', 'Box plot').to(
+    drag('[data-test="card-title"]', 'Box plot').to(
       '.grid-row.background--transparent:last',
     );
 
     // add back to dashboard
-    cy.get('.grid-container .box_plot').should('be.exist');
+    cy.get('[data-test="grid-container"]')
+      .find('.box_plot')
+      .should('be.visible');
 
     // should show Save changes button
-    cy.get('.dashboard-header .button-container').contains('Save');
+    cy.get('[data-test="header-save-button"]').should('be.visible');
 
     // undo 2 steps
-    cy.get('.dashboard-header .undo-action').click().click();
+    cy.get('[data-test="undo-action"]').click();
```

Comment:
Not sure why the second `cilck()` was removed here, but the comment above should be updated if this is intended.

More importantly, it's odd that there's no assertion here, like: 
• making sure the box_plot chart is gone again, after clicking the "undo" button.
• making sure the 'save' button is disabled after undoing the changes.

Reply:
added assertions and returned click back